CMakeLists.txt

```diff
@@ -0,0 +1,88 @@
+#Package information
+hyro_init_package(signal_generator VERSION 0.0.1)
+
+#Include packages of QtDesigner
+include ( FindPkgConfig  )
+find_package ( Qt4 COMPONENTS QtCore QtGui QtOpenGl )
+find_package ( QWT )
+
+#Required packages
+hyro_find_package(hyro_utils REQUIRED)
+hyro_find_package(hyro_msgs REQUIRED)
+hyro_find_package(hyro_widgets REQUIRED)
+
+include(${QT_USE_FILE} )
+add_definitions(${QT_DEFINITIONS})
+
+
+include_directories(
+
+  ${PROJECT_SOURCE_DIR}/src
+
+)
+
+#Declare header files
+set(SIGNAL_GENERATOR_HEADERS
+  include/hyro/SignalGeneratorComponent.h
+  include/hyro/DigitalConverterComponent.h
+  include/hyro/SignalGenerator.h
+  include/hyro/msgs/Signal.h
+  include/hyro/msgs/Signal.proto.h
+  include/hyro/Thresholding.h
```

Comment:
In the exercise, the you should declare `Thresholding.h` and `SignalGenerator.h` inside the folder `src/internal`.

Reply:
It is in src/internal